lib/tapioca/dsl/compilers/json_api_client_resource.rb

```diff
@@ -147,7 +147,17 @@ def type_for(property)
           return "T.untyped" if type.nil?
 
           sorbet_type = if type.respond_to?(:sorbet_type)
+            line, file = type.method(:sorbet_type).source_location
```

Comment:
This assumes that the `source_location` is available (i.e. this was defined in Ruby, and not in a C extension). Is that safe to assume, or should I guard against it?

Doing so is a bit uglier:

```ruby
msg = <<~MESSAGE
  WARNING: `#sorbet_type` is deprecated. Please rename your method to `#__tapioca_type`."
MESSAGE

if (source_location = type.method(:sorbet_type).source_location)
  msg.concat("Defined on line #{line} of #{file}"
end

msg = $stderr.puts
```